src/ws/service/friendships_service.rs

```diff
@@ -1,42 +1,70 @@
 use std::sync::Arc;
 
+use dcl_rpc::stream_protocol::Generator;
+
 use crate::{
-    ws::app::MyExampleContext, Empty, RequestEvents, ServerStreamResponse,
-    SharedFriendshipsService, SubscribeFriendshipEventsUpdatesResponse, UpdateFriendshipPayload,
-    UpdateFriendshipResponse, Users,
+    entities::friendships::FriendshipRepositoryImplementation, ws::app::MyContext, Empty,
+    RequestEvents, ServerStreamResponse, SharedFriendshipsService,
+    SubscribeFriendshipEventsUpdatesResponse, UpdateFriendshipPayload, UpdateFriendshipResponse,
+    User, Users,
 };
 
 pub struct MyFriendshipsService {}
 
 #[async_trait::async_trait]
-impl SharedFriendshipsService<MyExampleContext> for MyFriendshipsService {
+impl SharedFriendshipsService<MyContext> for MyFriendshipsService {
     async fn get_friends(
         &self,
         _request: Empty,
-        _context: Arc<MyExampleContext>,
+        context: Arc<MyContext>,
     ) -> ServerStreamResponse<Users> {
-        todo!()
+        // Get user_id from somewhere in the ether
+        let user_id = "";
+
+        // Look for friendships and build friend addresses list
+        let friendships = match &context.db.db_repos {
+            Some(repos) => {
+                let (friendships, _) = repos
+                    .friendships
+                    .get_user_friends(user_id, true, None)
```

Comment:
maybe we could try here to make a stream query in SQL? According to https://github.com/launchbadge/sqlx removing the `.await` in the `fetchAll` of the `get_user_friends` may do the trick